### PR TITLE
Implemented LegacyTransaction

### PR DESCRIPTION
**Pull request by jimni1222**
## Proposed changes

This PR introduces implementation of TxTypeLegacyTransaction.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-js/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-js)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

refer #249 

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -60,6 +61,8 @@ function Caver(provider, net) {
     this.account = Account
     this.wallet = new Wallet()
 
+    this.transaction = Transaction
+
     // ex) call `onit.klay.property` || `onit.klay.method(...)`
```
Comment (kjhman21) [inline index.js:66]:
```suggestion
    // ex) call `caver.klay.property` || `caver.klay.method(...)`
```

File: packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+const _ = require('lodash')
+const RLP = require('eth-lib/lib/rlp')
+const Bytes = require('eth-lib/lib/bytes')
+const AbstractTransaction = require('../abstractTransaction')
+const { TX_TYPE_STRING } = require('../../transactionHelper/transactionHelper')
+const utils = require('../../../../caver-utils/src')
+
+/**
+ * Representing a legacy transaction.
```
Comment (kjhman21) [inline packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js:27]:
```suggestion
 * Creates a legacy transaction.
```

File: packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+     * @override
+     * @return {string}
+     */
+    getRLPEncodingForSignature() {
+        if (this.gasPrice === undefined)
+            throw new Error(`gasPrice is undefined. Define variable in transaction or use 'transaction.fillTransaction' to fill values.`)
+        if (this.nonce === undefined)
+            throw new Error(`nonce is undefined. Define variable in transaction or use 'transaction.fillTransaction' to fill values.`)
+        if (this.chainId === undefined)
+            throw new Error(`chainId is undefined. Define variable in transaction or use 'transaction.fillTransaction' to fill values.`)
```
Comment (kjhman21) [inline packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js:172]: I think these error checks should be also checked in getRLPEncodingForTransactionHash(). What do you think?

Comment (jimni1222) [reply packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js:172]: Um... i didn't want to check optional values in method defined in each transaction types.
But for correct validation, i think those should be validated in each RLP functions.
I will edit! Thank you !

Comment (jimni1222) [reply packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js]: I modified it :)

File: packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+const _ = require('lodash')
+const RLP = require('eth-lib/lib/rlp')
+const Bytes = require('eth-lib/lib/bytes')
+const AbstractTransaction = require('../abstractTransaction')
+const { TX_TYPE_STRING, validateOptionalValues } = require('../../transactionHelper/transactionHelper')
+const utils = require('../../../../caver-utils/src')
+
+/**
+ * Creates a legacy transaction.
```
Comment (kjhman21) [inline packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js:27]:
```suggestion
 * Represents a legacy transaction.
```

File: packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+            gas: utils.trimLeadingZero(gas),
+            to,
+            value: utils.trimLeadingZero(value),
+            input: utils.trimLeadingZero(input),
+            signatures: [v, r, s],
+        })
+    }
+
+    /**
+     * Represents a legacy transaction.
```
Comment (kjhman21) [inline packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js:46]:
```suggestion
     * Creates a legacy transaction.
```

**Comment by jimni1222** [comment]
@kjhman21 @sirano11 
Dear reviewers, I did force push for rebasing on dev branch. ;-;
Please review when you have time.

**Pull request merged**